api-module/src/main/java/org/devridge/api/domain/member/service/MemberService.java

```diff
@@ -0,0 +1,119 @@
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+
+@Service
+@RequiredArgsConstructor
+public class MemberService {
+    private final MemberRepository memberRepository;
+    private final SkillRepository skillRepository;
+    private final MemberSkillRepository memberSkillRepository;
+    private final PasswordChecker passwordChecker;
+    private final BCryptPasswordEncoder passwordEncoder;
+
+    @Transactional
+    public void createMember(CreateMemberRequest reqDto){
+        checkDuplMember(reqDto);
+        passwordChecker.checkWeakPassword(reqDto.getPassword());
+        
+        if(!areSkillsValid(reqDto.getSkills())){
+            throw new SkillsNotValidException();
+        }
+
+        Member member = createNormalMember(reqDto);
+        createMemberSkill(reqDto, member);
+    }
+
+    @Transactional
+    public void createMemberSkill(CreateMemberRequest reqDto, Member member) {
+        Set<String> userSkills = new HashSet<>(List.of(reqDto.getSkills()));
+
+        for (String userSkill : userSkills) {
+            Optional<Skill> bySkill = skillRepository.findBySkill(userSkill);
+        }
+
+        /**
+         * 성능 최적화 하기!! (SQL query)
+        * */
+        for (String userSkill : userSkills) {
+            Skill skill = skillRepository.findBySkill(userSkill).get();
+            MemberSkillId memberSkillId = new MemberSkillId(member.getId(), skill.getId());
+
+            MemberSkill memberSkill = MemberSkill.builder()
+                    .id(memberSkillId)
+                    .member(member)
+                    .skill(skill)
+                    .build();
+            memberSkillRepository.save(memberSkill);
+        }
+
+    }
+
+    private void checkDuplMember(CreateMemberRequest reqDto) {
+        final Optional<Member> user = memberRepository.findByEmailAndProvider(reqDto.getEmail(), reqDto.getProvider());
+
+        if (user.isPresent()) {
+            throw new DuplEmailException("[ERROR] 이미 존재하는 계정입니다.");
+        }
+    }
+
+    private Member createNormalMember(CreateMemberRequest reqDto) {
+        System.out.println("=== createNormalMember() = createMember() ");
+        String encodedPassword = passwordEncoder.encode(reqDto.getPassword());
+
+        Member member = Member.builder()
+                .email(reqDto.getEmail())
+                .password(encodedPassword)
+                .provider(MemberConstant.PROVIDER_NORMAL)
+                .roles("ROLE_" + SecurityConstant.USER_ROLE)
+                .nickname(reqDto.getNickname())
+                .build();
+
+        return memberRepository.save(member);
+    }
+
+    public void deleteMember(DeleteMemberRequest reqDto) {
+        Member member = SecurityContextHolderUtil.getMember();
+
+        if (passwordEncoder.matches(reqDto.getPassword(), member.getPassword())){
```

Comment:
객체지향적인 코드에서 else 문을 사용하는 것은 지양하고 있는 편이긴 합니다. 아래와 같이 수정하는 것은 어떠신가요?
```suggestion
        if (!passwordEncoder.matches(reqDto.getPassword(), member.getPassword())) {
           throw new PasswordNotMatchException();
        }
        
        // 아래에 이후 로직 작성
```

Reply:
네 그게 더 깔끔할 것 같네요! 조언 감사합니다.
그렇게 바꾸겠습니다~!

Reply:
수정했습니다!